craft_providers/bases/buildd.py

```diff
@@ -100,6 +122,33 @@ def __init__(
 
         self.hostname = hostname
 
+    def _ensure_instance_config_compatible(
```

Comment:
Maybe this could be `_check_instance_config_compatibility`? "Ensure" suggests that this method will make the configuration be compatible if it's not.

Reply:
I do not have the same interpretation of ensure, but regardless this is a pattern I use fairly regularly. I'll ask the team if there is a preferred approach and update it in the future, if needed.